firebase-web/src/main/java/io/spine/web/firebase/NodeValue.java

```diff
@@ -32,25 +32,34 @@
 import com.google.firebase.database.utilities.DefaultClock;
 import com.google.firebase.database.utilities.OffsetClock;
 import com.google.gson.JsonObject;
+import com.google.protobuf.Message;
+import io.spine.annotation.Internal;
+
+import javax.annotation.Nullable;
 
 import static com.google.api.client.http.ByteArrayContent.fromString;
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.net.MediaType.JSON_UTF_8;
 import static com.google.firebase.database.utilities.PushIdGenerator.generatePushChildName;
+import static io.spine.json.Json.fromJson;
 
 /**
  * The Firebase database node value.
  */
+@Internal
 public final class NodeValue {
 
     private final JsonObject value;
+    @Nullable
+    private final StoredJson originalJson;
 
-    private NodeValue(JsonObject value) {
+    private NodeValue(JsonObject value, @Nullable StoredJson originalJson) {
```

Comment:
Please document the ctor. It's now a bit odd to see the value _and_ the original JSON out of the context of any operation.